games-emulation/ppsspp/ppsspp-1.11.3_p20210627.ebuild

```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake desktop xdg
+
+DESCRIPTION="A PSP emulator written in C++"
+HOMEPAGE="https://www.ppsspp.org/"
+COMMIT="ff2a6eeda2d00210f1a1f7bc0ab0ad99e7bc60b6"
+SRC_URI="
+	https://github.com/hrydgard/${PN}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz
+	https://github.com/KhronosGroup/SPIRV-Cross/archive/9acb9ec31f5a8ef80ea6b994bb77be787b08d3d1.tar.gz -> ${P}-ext_SPIRV-Cross.tar.gz
+	https://github.com/Kingcom/armips/archive/7885552b208493a6a0f21663770c446c3ba65576.tar.gz -> ${P}-ext_armips.tar.gz
+	https://github.com/Tencent/rapidjson/archive/73063f5002612c6bf64fe24f851cd5cc0d83eef9.tar.gz -> ${P}-ext_rapidjson.tar.gz
+	https://github.com/facebook/zstd/archive/56421f34e4ab5c17cbac0832bd12413e4f27ecf9.tar.gz -> ${P}-ext_zstd.tar.gz
+	https://github.com/hrydgard/glslang/archive/dc11adde23c455a24e13dd54de9b4ede8bdd7db8.tar.gz -> ${P}-ext_glslang.tar.gz
+	https://github.com/hrydgard/miniupnp/archive/3a87be33e797ba947b2b2a5f8d087f6c3ff4d93e.tar.gz -> ${P}-ext_miniupnp.tar.gz
+	https://github.com/hrydgard/ppsspp-freetype/archive/cbea79dc8fef4d9210e2bac7e7b9b5ff3388197a.tar.gz -> ${P}-ext_native_tools_prebuilt.tar.gz
+	https://github.com/hrydgard/ppsspp-lang/archive/3bf6b18eb0ae68861fbfbb7b43cd764771ca42a9.tar.gz -> ${P}-assets_lang.tar.gz
+	!system-ffmpeg? ( https://github.com/hrydgard/ppsspp-ffmpeg/archive/0b28335acea4f429ae798c5e75232e54881bf164.tar.gz -> ${P}-ffmpeg.tar.gz )
+	discord? ( https://github.com/discordapp/discord-rpc/archive/963aa9f3e5ce81a4682c6ca3d136cddda614db33.tar.gz -> ${P}-ext_discord-rpc.tar.gz )
+"
+
+RESTRICT="test"
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="discord qt5 sdl +system-ffmpeg"
+REQUIRED_USE="!qt5? ( sdl )"
```

Comment:
> Currently it also depends on libsdl2 even though it's not used when qt5 is set

You mean the code or ebuild?

On that note I looked into the codebase and it seems it depends on sdl while build with Qt.
So should it be done like this?:

```
RDEPEND="
    qt5? (
       ...
       media-libs/libsdl2[joystick]
   )
   !qt5? (
         media-libs/libsdl2[joystick,opengl,sound,video]
     )
"
```


```
elseif(USING_QT_UI)
        ...
        
        # Enable SDL joystick if SDL is found
        if(SDL2_FOUND)
                add_definitions(-DSDL)
                set(nativeExtra ${nativeExtra}
                        SDL/SDLJoystick.h
                        SDL/SDLJoystick.cpp
                )
                set(nativeExtraLibs ${nativeExtraLibs} SDL2::SDL2)
        endif()
...
```

Reply:
Oh, it still use it for joystick.

Does this mean joystick don't work if USE=sdl is not set? Maybe IUSE=+joystick would make more sense for this.

Reply:
I might add it's a global flag, so it's cheap to have and makes it clear won't have joystick if unset -- and still allows a dependency to be optional (but I'd make it default enabled).

Reply:
> I might add it's a global flag, so it's cheap to have and makes it clear won't have joystick if unset -- and still allows a dependency to be optional (but I'd make it default enabled).

If we do not modify CMakeLists.txt if a user disables joystick and has sdl they will have a failed build

Reply:
Well, *could* have a `REQUIRED_USE="!qt5? ( joystick )"` so that if sdl build is enabled, need joystick.

But well, given other comment I'll agree it's not worth it :) So yeah, let's just make sdl2[joystick] unconditionally required as you suggested.